### PR TITLE
Layer Integration

### LINKED ISSUE #14179: Add Layer Logger
## 🚀 Feature

Create LayerLogger to enable PL developer to track their ML experiments with [Layer](https://layer.ai)

### Motivation

Layer is an collaboration-first ML metadata store which provides tracking all ML entities from training data, ML models to experiments and we want to bring these functionalities into Pytorch Lightning Developers.

_Disclaimer: I'm the CEO and Founder of Layer_

### Pitch

Once LayerLogger (PR is on the way) integrated, we will provide PL Developers to:
- Log anything (images, videos, torch tensors, hyperparams etc) to their experiments.
- Version and log PL models easily while Layer takes care of serializing/deserializing it.
- Compare metrics between different PL model versions.

### Alternatives

There are existing experiment tracking tools in the market, integrated with PL already like; mlflow, Neptune, WandB etc.

### Additional context

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by mecevit**
## What does this PR do?

This PR introduces [Layer](https://layer.ai) integration as a Pytorch Lightning Logger. PL Developers can utilize Layer to track their experimentation through Layer Projects.

Fixes #14179

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

For sure, I did!!

Thanks to @fcakyon for his contributions and valuable feedback!

### CODE REVIEW

**Review by justusschock — CHANGES_REQUESTED** [review]
Blocking due to https://github.com/Lightning-AI/lightning/issues/14179#issuecomment-1214383080
